### ISSUE TITLE
Login with Apple ID

### ISSUE THREAD

**Issue by MoeNick** [GeneralMagicio/pairwise-RPGF4#9]

**Comment by MoeNick** [comment]
@mmahdigh @alireza-sharifpour success criteria's updated.

**Comment by maryjaf** [comment]
I couldn't test the sign flow with apple id , it should be tested by a user that has enabled two factor authenticator.

**Comment by MoeNick** [comment]
I asked Mohammad to test according to him:

I can successfully login but get redirected to this page again:
<img width="369" src="https://api.zenhub.com/attachedFiles/eyJfcmFpbHMiOnsibWVzc2FnZSI6IkJBaHBBOW4vQkE9PSIsImV4cCI6bnVsbCwicHVyIjoiYmxvYl9pZCJ9fQ==--33c7e7080804581d9d57e8a6f0e25c88e655f39e/image.png" alt="image.png" />

**Comment by MoeNick** [comment]
Some improvements needed for error handling @mmahdigh as we discussed.

**Comment by mmahdigh** [comment]
> Some improvements needed for error handling @mmahdigh as we discussed.

This is ready to test.

**Comment by maryjaf** [comment]
Please take a look on below screen record 
The login page is shown for a second, is this accepted ? @MoeNick @mmahdigh   
https://github.com/GeneralMagicio/pairwise-RPGF4/assets/111529185/419b16e5-fac7-4992-bc5a-dcd451c3cad0

**Comment by mmahdigh** [comment]
> Please take a look on below screen record The login page is shown for a second, is this accepted ? @MoeNick @mmahdigh
> 
>  Screen.Recording.2024-05-23.at.11.49.05.AM.mov

There is no perfect around this issue though.

The only thing we can do is to go to the "Account verified" page **before** signing in with Google/Apple then if the user fails to sign in, we'll have half a second of the "Account verified" page, and then the standard login page.

**Comment by MoeNick** [comment]
I guess it's ok, but if showing the login pages gets timely during the process (more than 0.5 sec ) , having a loading icon on that screen may help. @mmahdigh @maryjaf 
WDYT?

**Comment by mmahdigh** [comment]
> I guess it's ok, but if showing the login pages gets timely during the process (more than 0.5 sec ) , having a loading icon on that screen may help. @mmahdigh @maryjaf WDYT?

No, it won't. That wait is literally for getting back from the Thirdweb authentication window, our app registering that the login has happened and then going to the "Account verified" page.

There's no request or async operation happening there that would delay it further.

**Comment by mmahdigh** [comment]
> Yeah seem fair. But we can detect it's coming back from thirdweb don't we? so if we can do, we can add a loading to the login page only in this case. right?

Yeah, that's doable. Will get back to you when I add it.

**Comment by mmahdigh** [comment]
@MoeNick @maryjaf Added a loading message for the time users are logging in with their social accounts. Please retest it.
